File: frontend/types/ausencias.ts
export interface TipoAusencia {
  id: string
  nombre: string
  es_base: boolean
  activo: boolean
}

export interface TipoAusenciaListResponse {
  items: TipoAusencia[]
  total: number
}

export interface Ausencia {
  id: string
  empresa_id: string
  empresa_nombre: string | null
  empleado_id: string
  empleado_nombre: string | null
  area_id: string | null
  area_nombre: string | null
  tipo_id: string
  tipo_nombre: string | null
  fecha_desde: string  // ISO "YYYY-MM-DD"
  fecha_hasta: string
  dias: number
  justificada: boolean
  motivo: string | null
  created_at: string
}

export interface AusenciaCreate {
  empleado_id: string
  tipo_id: string
  fecha_desde: string
  fecha_hasta: string
  justificada: boolean
  motivo?: string
}

export interface AusenciaUpdate {
  tipo_id?: string
  fecha_desde?: string
  fecha_hasta?: string
  justificada?: boolean
  motivo?: string
}

export interface AusenciaListResponse {
  items: Ausencia[]
  total: number
}

File: backend/schemas/ausencias.py
"""
Schemas Pydantic para el módulo de ausencias.
TipoAusencia: catálogo global (sin empresa_id).
Ausencia: Create → Update → Response → ListResponse.
empresa_id en Response es HEREDADO del empleado al crear, nunca lo provee el usuario.
"""
from datetime import date, datetime
from typing import List, Optional
from uuid import UUID

from pydantic import BaseModel


class TipoAusenciaCreate(BaseModel):
    nombre: str


class TipoAusenciaResponse(BaseModel):
    id: str
    nombre: str
    es_base: bool
    activo: bool


class TipoAusenciaListResponse(BaseModel):
    items: List[TipoAusenciaResponse]
    total: int


class AusenciaCreate(BaseModel):
    empleado_id: UUID
    tipo_id: UUID
    fecha_desde: date
    fecha_hasta: date
    justificada: bool = False
    motivo: Optional[str] = None


class AusenciaUpdate(BaseModel):
    tipo_id: Optional[UUID] = None
    fecha_desde: Optional[date] = None
    fecha_hasta: Optional[date] = None
    justificada: Optional[bool] = None
    motivo: Optional[str] = None


class AusenciaResponse(BaseModel):
    id: str
    empresa_id: str
    empresa_nombre: Optional[str] = None
    empleado_id: str
    empleado_nombre: Optional[str] = None
    area_id: Optional[str] = None
    area_nombre: Optional[str] = None
    tipo_id: str
    tipo_nombre: Optional[str] = None
    fecha_desde: date
    fecha_hasta: date
    dias: int
    justificada: bool
    motivo: Optional[str] = None
    created_at: datetime


class AusenciaListResponse(BaseModel):
    items: List[AusenciaResponse]
    total: int
